packages/nimble-components/src/table/models/keyboard-navigation-manager.ts

```diff
@@ -0,0 +1,1025 @@
+            }
+        } else {
+            const headerActionIndex = this.focusableHeaderElements.headerActions.indexOf(
+                activeElement
+            );
+            if (shiftKeyPressed) {
+                if (headerActionIndex > -1) {
+                    if (headerActionIndex > 0) {
+                        this.focusState.focusType = TableFocusType.headerActions;
+                        this.focusState.headerActionIndex = headerActionIndex - 1;
+                        focusableElement = this.focusableHeaderElements.headerActions[
+                            this.focusState.headerActionIndex
+                        ];
+                    }
+                } else if (
+                    this.focusableHeaderElements.columnHeaders.includes(
+                        activeElement
+                    )
+                ) {
+                    if (this.focusableHeaderElements.headerActions.length > 0) {
+                        this.focusState.focusType = TableFocusType.headerActions;
+                        this.focusState.headerActionIndex = this.focusableHeaderElements.headerActions.length
+                            - 1;
+                        focusableElement = this.focusableHeaderElements.headerActions[
+                            this.focusState.headerActionIndex
+                        ];
+                    }
+                }
+            } else if (headerActionIndex > -1) {
+                if (
+                    headerActionIndex + 1
+                    < this.focusableHeaderElements.headerActions.length
+                ) {
+                    this.focusState.focusType = TableFocusType.headerActions;
+                    this.focusState.headerActionIndex = headerActionIndex + 1;
+                    focusableElement = this.focusableHeaderElements.headerActions[
+                        this.focusState.headerActionIndex
+                    ];
+                }
+            }
+        }
+        if (focusableElement) {
+            this.inNavigationMode = this.focusState.focusType !== TableFocusType.cellContent
+                && this.focusState.focusType !== TableFocusType.cellActionMenu;
+            this.focusElement(focusableElement);
+            return true;
+        }
+        this.blurAfterLastTab(activeElement);
+        return false;
+    }
+
+    private blurAfterLastTab(activeElement: HTMLElement): void {
+        // In order to get the desired browser-provided Tab/Shift-Tab behavior of focusing the
+        // element before/after the table, the table shouldn't have tabIndex=0 when this event
+        // handling ends. However it needs to be tabIndex=0 so we can re-focus the table the next time
+        // it's tabbed to, so set tabIndex back to 0 after a rAF.
+        // Note: In Chrome this is only needed for Shift-Tab, but in Firefox both Tab+Shift-Tab need this
+        // to work as expected.
+        this.table.tabIndex = -1;
+        window.requestAnimationFrame(() => {
```

Comment:
`requestAnimationFrame` calls always need vetting/questioning, so opening a comment thread here. See code comments for why this is here (if the table was still tabindex=0 when the user did Shift-Tab, the overall table would get focused which isn't what we want).

I'm also not aware of a good way to determine what the next or previous focused element will be, if we wanted to focus it programmatically (the browser computes it based on semi-complicated tab order rules).  
So this code is just trying to ensure the browser can handle it: we blur the focused table sub-element, and then we don't `preventDefault()` on the Tab/Shift-Tab key event, so the browser will do the default behavior still (of focusing the next/previous element in tab order).